core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -388,11 +412,31 @@ class LogManager(val logDirs: Array[File],
         cleaner.abortCleaning(topicAndPartition)
         cleaner.updateCheckpoints(removedLog.dir.getParentFile)
       }
-      removedLog.delete()
-      info("Deleted log for partition [%s,%d] in %s."
-           .format(topicAndPartition.topic,
-                   topicAndPartition.partition,
-                   removedLog.dir.getAbsolutePath))
+      // renaming the directory to topic-partition.uniqueId-delete
+      val dirName = StringBuilder.newBuilder
+      dirName.append(removedLog.name)
+      dirName.append(".")
+      val uniqueId = java.util.UUID.randomUUID.toString.replaceAll("-","")
+      dirName.append(uniqueId)
+      dirName.append(Log.DeleteDirSuffix)
+      removedLog.close()
```

Comment:
One important point to note is that this involves a flush. If we are deleting several topics and there are several partitions on this broker then this is a potential blocking point (say order of seconds). I don't think we need to address this immediately but I think it can and should be addressed at some point to make the delete truly async.